R/QFeatures-join.R

```diff
@@ -149,14 +152,20 @@ mergeSElist <- function(x) {
 ##' rowData(feat2[["joinedAssay"]])
 joinAssays <- function(x,
                        i,
-                       name = "joinedAssay") {
+                       name = "joinedAssay",
+                       fcol = NULL) {
     stopifnot("Object must be of class 'QFeatures'" = inherits(x, "QFeatures"),
               "Need at least 2 assays to join" = length(i) >= 2)
     if (name %in% names(x))
         stop("Assay with name '", name, "' already exists.")
+    if (!is.null(fcol)) {
+        ## Set assay rownames for joining
+        message("Using '", fcol, "' to join assays.")
+        x <- .setAssayRownames(x, fcol)
+    }
     ## Join assays and add to x
     joined_se <- mergeSElist(experiments(x)[i])
     x <- addAssay(x, joined_se, name = name)
     ## Add the multi-parent AssayLinks
     addAssayLink(x, from = i, to = name)
```

Comment:
I see a third option: use the `varFrom` and the `varTo` arguments of `addAssayLink()` to pass `fcol` as the linking variable between the joined sets. I think that should work, although I didn't test it.